### ISSUE TITLE
M4Bs modified by Tone 0.1.6+ will not show when played in mpv

### ISSUE THREAD

**Issue by 99bugsleft** [sandreas/tone#79]
Hello. I am not very familiar with audio formats, so please bear with me. If this is a bug in mpv and not tone, please close the ticket.

I use debian based. x64.
mpv:
```
mpv 0.37.0 Copyright © 2000-2023 mpv/MPlayer/mplayer2 projects
libplacebo version: v6.338.2
FFmpeg version: 6.1.1
FFmpeg library versions:
- libavutil=58.29.100
- libavcodec=60.31.102
- libavformat=60.16.100
- libswscale=7.5.100
- libavfilter=9.12.100
- libswresample=4.12.100
```
Tone version works with mpv:
- 0.1.4
- 0.1.5

Tone version does not work with mpv:
- 0.1.6
- 0.1.8
- 0.2.1
- 0.2.2
- 0.2.3

I did not test 0.1.7, 0.1.9, and 0.2.0.

I use tone to add metadata to my m4b files. Today, I checked if there is a new version as I had been using v1.4 for some time. I used 2.3 to modify my audio. Afterwards, it would still play in mpv but the window would be hidden/cannot be seen/there is no GUI.

ffprobe shows this:
```
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'redact_fix.m4b':
  Metadata:
    major_brand     : M4A 
    minor_version   : 512
    compatible_brands: M4A isomiso2
    title           : Linux
    artist          : anon
    album           : Bestof
    comment         : V
    genre           : Audiobook
    date            : 2024-01-01
    encoder         : Lavf60.16.100
    media_type      : 2
    gapless_playback: 1
  Duration: 08:10:12.47, start: 0.000000, bitrate: 64 kb/s
  Chapters:
    Chapter #0:0: start 0.000000, end 545.620000
[clip]
  Stream #0:0[0x1](und): Audio: aac (HE-AAC) (mp4a / 0x6134706D), 44100 Hz, stereo, fltp, 64 kb/s (default)
    Metadata:
      handler_name    : SoundHandler
      vendor_id       : [0][0][0][0]
  Stream #0:1[0x2](und): Data: bin_data (text / 0x74786574) (default)
    Metadata:
      creation_time   : 2024-12-17T01:26:58.000000Z
      handler_name    : Chapter titles
  Stream #0:2[0x3](und): Video: mjpeg (Baseline) (jpeg / 0x6765706A), gray(bt470bg/unknown/unknown), 1x1 [SAR 1:1 DAR 1:1], 0 kb/s, 0.0012 fps, 1k tbr, 1k tbn (default)
    Metadata:
      creation_time   : 2024-12-17T01:26:58.000000Z
      handler_name    : Chapter pictures
      vendor_id       : [0][0][0][0]
  Stream #0:3[0x0]: Video: mjpeg (Baseline), yuvj420p(pc, bt470bg/unknown/unknown), 400x400 [SAR 1:1 DAR 1:1], 90k tbr, 90k tbn (attached pic)
```

If I do this, the file opens as expected again:
`ffmpeg -i "linux.m4b" -map 0:0 -c copy "linux_fix.m4b"`
But it removes some of the metadata. 

If I use tone 0.1.4 or 0.1.5, no new stream is created when changing the metadata:
```
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'redact_fix.m4b':
  Metadata:
    major_brand     : M4A 
    minor_version   : 512
    compatible_brands: M4A isomiso2
    title           : Linux
    artist          : anon
    album           : Bestof
    comment         : V
    genre           : Audiobook
    date            : 2024-01-01
    encoder         : Lavf60.16.100
    media_type      : 2
    gapless_playback: 1
  Duration: 08:10:12.47, start: 0.000000, bitrate: 64 kb/s
  Chapters:
    Chapter #0:0: start 0.000000, end 545.620000
[clip]
  Stream #0:0[0x1](und): Audio: aac (HE-AAC) (mp4a / 0x6134706D), 44100 Hz, stereo, fltp, 64 kb/s (default)
    Metadata:
      handler_name    : SoundHandler
      vendor_id       : [0][0][0][0]
  Stream #0:1[0x2](und): Data: bin_data (text / 0x74786574) (default)
    Metadata:
      creation_time   : 2024-12-17T02:58:46.000000Z
      handler_name    : Chapter titles
```

As tone 0.1.5 does everything I need, I will just use the old version for now, but I am curious what the issue is.
If you need more data, I am happy to provide. Thank you for developing this amazing tool!

**Comment by sandreas** [comment]
I'm going to take a look. Could you provide a non copyrighted example file  and the commands to reproduce this, would save me a lot of time.

**Comment by sandreas** [comment]
@99bugsleft 

Ok, I reproduced it with one file, but it's not that easy. Usually I use the following commands to reproduce it on a non-copyrighted file:

```bash
# create a 5 seconds sample silence
ffmpeg -ar 48000 -ac 1 -f s16le -i /dev/zero -t 5 -y sample.wav

# convert wav to m4b
ffmpeg -i sample.wav -f mp4 sample.m4b

# download a sample cover
wget -c https://picsum.photos/id/237/500/500 -O cover.jpg

# embed the cover into the sample
ffmpeg -i sample.m4b -i cover.jpg -map 0 -map 1 -c copy -disposition:v:1 attached_pic -f mp4 ready.m4b
```

Now, if I use tone to tag this file, everything works flawless.
However, I have one file that shows exactly the same behaviour as yours, but it is under copyright - but I try to reproduce it with a non copyrighted one, just takes slightly more time than expected.

**Comment by sandreas** [comment]
@99bugsleft 
Awesome thank you for taking the effort. I already submitted [an issue](https://github.com/Zeugma440/atldotnet/issues/296) in `atldotnet`, but your file helps A LOT to reproduce the problem.

I let you know as soon as I find out something, may take a while though.

**Comment by sandreas** [comment]
@99bugsleft 
So, this has been partially fixed for `0.2.4`. Partially means that there is still seems to be an issue in `mpv`, that leads to this behaviour (see https://github.com/Zeugma440/atldotnet/issues/296#issuecomment-2558490190 for details). 

However, I now consider this as fixed for `tone`. 

If you have another issue, feel free to reopen.
